src/test/cucumber/pre_and_post_installation_hooks.feature

```diff
@@ -7,26 +7,29 @@ Feature: Hooks
 	Scenario: Pre- and Post-installation Hooks return successfully
 		And a machine with "Linux" installed
 		And the system is bootstrapped
-		And the candidate "java" version "8u92" is available for download on "Linux"
-		And a "pre" install hook is served for "java" "8u92" on "Linux" that returns successfully
-		And a "post" install hook is served for "java" "8u92" on "Linux" that returns successfully
-		When I enter "sdk install java 8u92"
+		And the candidate "grails" version "2.1.0" is available for download on "Linux"
+		And a "pre" install hook is served for "grails" "2.1.0" on "Linux" that returns successfully
+		And a "post" install hook is served for "grails" "2.1.0" on "Linux" that returns successfully
+		When I enter "sdk install grails 2.1.0"
 		Then I see "Pre-installation hook success"
 		And I see "Post-installation hook success"
+		And the exit code is 0
 
 	Scenario: Pre-installation Hook returns a non-zero code
 		And a machine with "Linux" installed
 		And the system is bootstrapped
-		And the candidate "java" version "8u92" is available for download on "Linux"
-		And a "pre" install hook is served for "java" "8u92" on "Linux" that returns a failure
-		When I enter "sdk install java 8u92"
+		And the candidate "grails" version "2.1.0" is available for download on "Linux"
+		And a "pre" install hook is served for "grails" "2.1.0" on "Linux" that returns a failure
+		When I enter "sdk install grails 2.1.0"
 		Then I see "Pre-installation hook failure"
+		And the exit code is 1
 
 	Scenario: Post-install Hook returns a non-zero code
 		And a machine with "Linux" installed
 		And the system is bootstrapped
-		And the candidate "java" version "8u92" is available for download on "Linux"
-		And a "pre" install hook is served for "java" "8u92" on "Linux" that returns successfully
-		And a "post" install hook is served for "java" "8u92" on "Linux" that returns a failure
-		When I enter "sdk install java 8u92"
+		And the candidate "grails" version "2.1.0" is available for download on "Linux"
+		And a "pre" install hook is served for "grails" "2.1.0" on "Linux" that returns successfully
+		And a "post" install hook is served for "grails" "2.1.0" on "Linux" that returns a failure
+		When I enter "sdk install grails 2.1.0"
 		Then I see "Post-installation hook failure"
+		And the exit code is 1
```

Comment:
I believe Java 8u92 and 8.0.101 requires a custom post-installation script, not the default one provided by the Cucumber statement. So I changed to grails 2.1.0, which works. I believe this error was there but we weren't catching it before.

Reply:
The intent of the test was to ensure that pre and post hooks (for Java) are behaving properly. Sorry, I should have been a bit more explicit in the description of this test.

We already have two canned responses (success and failure) by way of `jdk-8.0.101` and `jdk-8.0.111` that should be used instead.

Let's fix this as a separate PR please. It really shouldn't take very long.

Reply:
I agree.